docs/feature_combo.md

```diff
@@ -67,31 +86,100 @@ void process_combo_event(uint8_t combo_index, bool pressed) {
+
+#### Long Combos
+If you're using long combos or longer combos, you may run into issues with this, as the structure may not be large enough to accommodate what you're doing. In this case, you can configure the size of the data structure used. Be aware, larger combo sizes will increase memory usage!
+
+| Keys | Define to be set     |
+|------|----------------------|
+| 8    | QMK Default          |
+| 16   | `#define EXTRA_LONG_COMBOS`  |
+| 32   | `#define EXTRA_EXTRA_LONG_COMBOS`|
+
+#### Modifier Combos
+If a combo resolves to a Modifier, the window for processing the combo can be extended independent from normal combos. By default this is disabled but can be enabled with `#define COMBO_MUST_HOLD_MODS`, and the time window can be configured with `#define COMBO_MOD_TERM 150` (default: 200). With `COMBO_MUST_HOLD_MODS` you cannot tap the combo any more which makes the combo less prone to misfires.
+
+#### Permissive Hold
+Similar to Tap-Hold configuration, Mod Combos can be setup to behave similarly. Add `#define COMBO_PERMISSIVE_HOLD` to your `config.h` to enable. This feature is extremely useful for fast typists as you don't have to wait for the Mod Combo's combo term to finish.
+
+If `COMBO_TERM` and `COMBO_MOD_TERM` are too close to each other this settings becomes less useful as this only affects if a key is pressed between `COMBO_TERM` and `COMBO_MOD_TERM`. All key presses under `COMBO_TERM` are always considered for overlapping combos.
+
+#### Per Combo Timing
+Instead of using a blanket `COMBO_TERM` window for every combo, per-combo timings can be generated. In order to use this feature the following config options and functions can be defined. Coming up with useful timings and configuration is left as an exercise for the reader.
+
+| Config Flag                 | Function                                                  | Description                                                                                          |
+|-----------------------------|-----------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `COMBO_MUST_HOLD_PER_COMBO` | bool get_combo_must_hold(uint16_t index, combo_t \*combo) | Controls if a given combo should fire immediately on tap or if it needs to be held. (default: False) |
+| `COMBO_TERM_PER_COMBO`      | uint16_t get_combo_term(uint16_t index, combo_t \*combo)  | Optional per-combo timeout window. (default: COMBO_TERM)                                             |
+
+If you set either of the options but like the Modifier Combo settings, you have to define those yourself in the function.
+
+Examples:
+```c
+uint16_t get_combo_term(uint16_t index, combo_t *combo) {
+    // decide by combo->keycode
+    if (IS_MOD(combo->keycode)) return COMBO_MOD_TERM; // you have to config this yourself ifdef COMBO_TERM_PER_COMBO
```

Comment:
Is the « ifdef COMBO_TERM_PER_COMBO » supposed to be in the comment? Or is in the sense of " you have to configure this yourself if you're using COMBO_TERM_PER_COMBO "?

Reply:
Yeah it was just that. :P I changed it so no one gets confused by it.